src/dcp/dcp.c

```diff
@@ -364,7 +364,21 @@ int main(int argc, char** argv)
         MPI_Finalize();
         return 1;
     }
-    
+   
+    /* check if source or destination is an HDF5 container, if that is
+     * the case then fork/exec h5repack */
+    rc = mfu_daos_hdf5_copy(mfu_src_file, mfu_dst_file, argpaths, daos_args);
+    if (rc != 0) {
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    } else {
+        /* h5repack was successful, parent process can return  */
+        mfu_finalize();
+        MPI_Finalize();
+        return 0;
+    }
```

Comment:
@dsikich We should only call this if either the source or dst is HDF5. Otherwise, we'll never get to the POSIX or DAOS copies. Also, maybe we can simplifythe if/else statement.
```suggestion
    if (mfu_src_file->type == HDF5 || mfu_dst_file->type == HDF5) {
        rc = mfu_daos_hdf5_copy(mfu_src_file, mfu_dst_file, argpaths, daos_args);
        if (rc != 0) {
            rc = 1;
        }
        mfu_finalize();
        MPI_Finalize();
        return rc;
    }
```